### ISSUE TITLE
Fig-align does not work if caption is empty or no tag is provided

### ISSUE THREAD

**Issue by ratnanil** [quarto-dev/quarto-cli#7528]
### Bug description

If a figure has either an empty caption or no id, then `width` is ignored in the pdf / typst output:

### Steps to reproduce

Render the following qmd with `quarto render index.qmd`

```qmd
---
format: typst
---

This figure should be center. It *isn't*.

![](./elephant.png){fig-align="center" width="20%"}

This figure should be center. It *is*.

![](./elephant.png){#fig-elephant fig-align="center" width="20%"}

This figure should be center. It *is*.

![Elephant](./elephant.png){fig-align="center" width="20%"}
``` 

### Expected behavior

I would expect all three images to look the same (centered)

### Actual behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/12532091/8a016aa6-3c8c-4d4d-9343-74c11568b385)

### Your environment

- IDE: VSCode
- OS: Ubuntu 22.04

### Quarto check output

```bash
quarto check                
Quarto 1.4.489
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.8: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.4: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.489
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.10
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/nils/.TinyTeX/bin/x86_64-linux
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: 5.3.1
      Kernels: ir, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by cderv** [comment]
Thanks for the report. I believe this also no applying to HTML  or PDF, and this is not specific to typst. 

This part of our doc seems to have a regression: https://quarto.org/docs/authoring/figures.html#figure-alignment

It works ok with 1.3

**Comment by cscheid** [comment]
> Thanks for the report. I believe this also no applying to HTML or PDF, and this is not specific to typst.

We need to be a little careful here, because `fig-align` is meant to work for _figures_. Figures have captions and reference IDs. This matters in 1.4, because not all images are figures, and not all figures are images.

I did just fix a related thing from [knitr behavior specifically](https://github.com/quarto-dev/quarto-cli/pull/7519), but I think we don't want to replicate the behavior across all of quarto's formats.

**Comment by cderv** [comment]
> We need to be a little careful here, because fig-align is meant to work for figures. Figures have captions and reference IDs. This matters in 1.4, because not all images are figures, and not all figures are images.

I understand new behavior  and how you expect it to be, though it was not working like in 1.3. `fig-align` could also be used to align an image without caption or with a caption and no id. 

Per our doc at: https://quarto.org/docs/authoring/figures.html#figure-alignment

````markdown
---
format:
  html: default
  pdf: default
---

# No caption

![](https://raw.githubusercontent.com/quarto-dev/quarto-web/main/docs/authoring/_figure-examples/elephant.png){fig-align="center" width="20%"}

# Caption No Id

![This is an elephant](https://raw.githubusercontent.com/quarto-dev/quarto-web/main/docs/authoring/_figure-examples/elephant.png){fig-align="center" width="20%"}


# Caption and fig id

![This is an elephant](https://raw.githubusercontent.com/quarto-dev/quarto-web/main/docs/authoring/_figure-examples/elephant.png){#fig-elephant fig-align="center" width="20%"}


See this beautilful elephant @fig-elephant

````

With 1.4 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/c15b1eee-c1b5-4ea0-af34-f75a3abeca0a)

With 1.3 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/b6637145-c0a1-41cb-84c8-fea6dfec0145)

This is why I added Regression. 

If this is expected change, then we need to warn about this for all user that were using `fig-align` on image with no caption. 


This is not related to **knitr** directly. **knitr** was just adding the `fig-align` attribute for the markdown syntax, that was working  before.

**Comment by cscheid** [comment]
Nope, I take it back. That's convincing to me, thanks for setting me straight!

(I still think we ought to be careful and document `layout-align` instead of `fig-align`, but we never want to regress.)

**Comment by cderv** [comment]
> (I still think we ought to be careful and document layout-align instead of fig-align, but we never want to regress.)

I am not sure I see the difference really but in the **knitr** currently we do set `layout-align` even when no fig- id or no caption. 

Which produce 
````markdown
::: {.cell layout-align="center"}

```{.r .cell-code}
library(ggplot2)
ggplot(mtcars, aes(x = cyl, y = mpg)) +
    geom_point()
```

::: {.cell-output-display}
![](index_files/figure-html/unnamed-chunk-1-1.png){fig-align='center' width=672}
:::
:::
````

So maybe this is where the issue was for #7519  and we should not produce this `layout-align` always 🤔 
https://github.com/quarto-dev/quarto-cli/blob/3aa0cb910aeaf8080ba7ba6322d502b6db87c5ad/src/resources/rmd/hooks.R#L273-L277

**Comment by cderv** [comment]
Ok good to know. So maybe we need to teach our R code for **knitr** to do the right thing then and use `fig-align` only when fig- label and/or a caption ?

**Comment by cscheid** [comment]
No, because that would be a regression :) I think we just need to document layout-align by default.

**Comment by cscheid** [comment]
PR is up for HTML and LaTeX fixes.

typst will be in a different PR to adjust Pandoc 3.1.9 changes.
